### ISSUE TITLE
Dev server compiles twice in VSCode

### ISSUE THREAD

**Issue by tavoyne** [webpack/webpack#15248]
# Bug report

**What is the current behavior?**

`webpack-dev-server` (via `webpack serve`) compiles twice on every save in VSCode. This doesn't happen when using an other editor such as Sublime.

**If the current behavior is a bug, please provide the steps to reproduce.**

Create a fresh package with `webpack`, `webpack-cli` and `webpack-dev-server` as `devDependencies`.

```javascript
// src/index.js

console.log("Hi");
```

Launch the server:
```bash
webpack serve --mode=production
```

Open the `src/index.js` file in VSCode. You'll notice in the console's output that every time you press save, the code gets compiled twice. Try with an other editor (such as Sublime) and you'll see that compilation occurs once, as expected.

**What is the expected behavior?**

Compilation should occur only once.

**Other relevant information:**
webpack version:  `v5.67.0`
Node.js version: `v16.13.0`
Operating System:  MacOS Monterey
Additional tools: VSCode `v1.63.2` (with default settings + no extensions)

**Comment by vankop** [comment]
I think this is duplicate of https://github.com/webpack/webpack/issues/15206 .

**Comment by tavoyne** [comment]
I don't think so, @vankop. First because downgrading doesn't fix the issue and it happens on VSCode not on Sublime.

**Comment by alexander-akait** [comment]
Check your save mode in `VSCode`, maybe your IDE create extra files on saving and trigger to run webpack, sorry I can't reproduce it locally

**Comment by tavoyne** [comment]
What do you mean by _save mode_? It occurs even with the default VSCode config (in which `formatOnSave` is disabled) and no extensions installed. I tried with a fresh install of VScode, still occurs.

**Comment by alexander-akait** [comment]
Can you provide stats output?

**Comment by tavoyne** [comment]
Sure! Here's what a single save causes:

![image](https://user-images.githubusercontent.com/29731597/151160764-34498262-81ee-4f80-984e-56d8b3b058f5.png)

**Comment by alexander-akait** [comment]
Do you have `webpack.config.js`? Sounds like something was changed in source

**Comment by tavoyne** [comment]
I'm using the default config here, so no `webpack.config.js` file, just the `webpack serve --mode=production` command.

**Comment by tavoyne** [comment]
The set-up is really minimal here. You can check: https://github.com/tavoyne/webpack-test.

**Comment by alexander-akait** [comment]
Sorry, I can't reproduce, it happens only in `VSCode`? Can you try other IDE/editor, just for testing

**Comment by tavoyne** [comment]
Sorry for the delay, I didn't see your reply.

Actually, I just tried on TextEdit (`v1.17`) and it happened there as well. But again, not on Sublime (`Build 4126`), surprisingly.

What function is used internally by `webpack-dev-server` to watch for file changes? Maybe it has something to do with that.

**Comment by alexander-akait** [comment]
Can you try `5.65.0` version?

**Comment by tavoyne** [comment]
YES, it fixes the problem. My bad, I did try downgrading before but to `v5.66.0` only... 

So this indeed is a #15206 duplicate.
